app/commands/build_blob_command.rb

```diff
@@ -5,6 +5,7 @@ def initialize(uuid:, async: true, **_args)
 
   def apply(root_object, meta:, **_args)
     meta[:cloud_provider] = Rails.application.config.active_storage.service.to_s
+    EventService.publish('BlobBuilt', root_object)
```

Comment:
You will see lots of these - the events are being fired in the commands now - more like CQRS - keeps this code out of the controllers and allows the command to have control of firing the event (with duplicate commands, the event should not be fired)